buildkite.sh

```diff
@@ -21,13 +11,57 @@ steps:
            tar xvfz master.tar.gz --strip-components=2 ci-master/buildkite
            rm -rf master.tar.gz && popd
         fi
+EOM
+
+cat << EOF
+steps:
+  - command: |
+        echo "--- Print environment"
+        uname -a
+        make --version
+        \\\${SHELL} --version || true
+        c++ --version
+        ld -v
+        ! command -v gdb &>/dev/null || gdb --version
+        ! dmd --version # ensure that no dmd is the current environment
+        ${LOAD_CI_FOLDER}
         ./buildkite/build_distribution.sh
     label: "Build"
     artifact_paths: "distribution.tar.xz"
     retry:
       automatic:
         limit: 2
+EOF
 
+################################################################################
+# Test bootstrapping with different compilers
+################################################################################
+
+case "${BUILDKITE_REPO:-x}" in
+    "https://github.com/dlang/dmd.git" | \
+    "https://github.com/dlang/druntime.git" | \
+    "https://github.com/dlang/phobos.git" | \
+    "https://github.com/dlang/ci.git")
+
+for line in dmd-64 gdc-64 ldc-64 ; do
+    # TODO: dmd-32
+    IFS=- read -r compiler model <<< "$line"
+cat << EOF
+  - command: |
+        ${LOAD_CI_FOLDER}
+        DMD=$compiler MODEL=$model ./buildkite/test_bootstrap.sh
+    label: "Bootstrap ($compiler)"
+EOF
+
+done
+    ;;
+    *)
+    ;;
+esac
+
+################################################################################
+
+cat << 'EOF'
   - wait
```

Comment:
So ideally we could start these tests in the beginning (they are totally independent to the distribution step and the following projects.)
However, it looks like a wait is a global one and thus we seem to only have two options:
- add them after the `wait` on the distribution
- add the bootstrapping projects as separate pipeline